File: python/word-count/word_count.py
import string

def word_count(text):
  
  words = {}
  
  cleaned_text = text.replace("_", " ").replace(",", " ").translate(None, string.punctuation).lower()
  
  for word in cleaned_text.split():
    if word in words:
      words[word] += 1
    else:
      words[word] = 1
      
  return words

File: r/word-count/word-count.R
word_count <- function(text) {

  # Get rid of punctuation, make lowercase, split
  words <- strsplit(tolower(gsub("[[:punct:]]", "", text)), " ")
  # Get rid of blanks, count, and transform into list
  as.list(table(lapply(words, function(x){x[!x ==""]})))
  
}
